src/api/app.py

```diff
@@ -215,7 +281,7 @@ def provision(request: Request) -> Redirect:
     )
 
 
-@get("/provision_handler", media_type=MediaType.HTML)
+@get("/provision_handler", media_type=MediaType.HTML, sync_to_thread=True)
 def provision_handler(request: Request) -> str:
```

Comment:
Function could be further broken up for readability (kind of nit-picky since this is not a part of this PR).

Reply:
yeah I gotta think about how to do that, I dont mind having logic specific to each end point in that endpoint as long as it isnt used anywhere else but I see what you mean

Reply:
I refactored some of this out into lib